yt_dlp/extractor/medici.py

```diff
@@ -1,67 +1,164 @@
+                'thumbnail': r're:^https?://medicitv-[abc]\.imgix\.net/movie/[^?]+\.jpg(?:\?[^?]+)?',
+                'description': 'md5:0f15a15611ed748020c769873e10a8bb',
+                'upload_date': '20240223',
+            },
+        },
+        {
+            'url': 'https://www.medici.tv/en/documentaries/la-sonnambula-liege-2023-documentaire',
+            'md5': '87ff198018ce79a34757ab0dd6f21080',
+            'expected_warnings': [r'preview'],
+            'info_dict': {
+                'id': '7513',
+                'ext': 'mp4',
+                'title': 'La Sonnambula',
+                'thumbnail': r're:^https?://medicitv-[abc]\.imgix\.net/movie/[^?]+\.jpg(?:\?[^?]+)?',
+                'description': 'md5:0caf9109a860fd50cd018df062a67f34',
+                'upload_date': '20231103',
+            },
+        },
+        {
+            'url': 'https://edu.medici.tv/en/masterclasses/yvonne-loriod-olivier-messiaen',
+            'md5': 'fb5dcec46d76ad20fbdbaabb01da191d',
+            'skip': 'The preview doesn\'t start from the start. Only works when authenticated.',
+            'info_dict': {
+                'id': '3024',
+                'ext': 'mp4',
+                'title': 'Olivier Messiaen and Yvonne Loriod, pianists and teachers',
+                'thumbnail': r're:^https?://medicitv-[abc]\.imgix\.net/movie/[^?]+\.jpg(?:\?[^?]+)?',
+                'description': 'md5:aab948e2f7690214b5c28896c83f1fc1',
+                'upload_date': '20150223',
+            },
+        },
+        {
+            'url': 'https://www.medici.tv/en/jazz/makaya-mccraven-la-rochelle',
+            'md5': '4cc279a8b06609782747c8f50beea2b3',
+            'expected_warnings': [r'preview'],
+            'info_dict': {
+                'id': '7922',
+                'ext': 'mp4',
+                'title': 'NEW: Makaya McCraven in La Rochelle',
+                'thumbnail': r're:^https?://medicitv-[abc]\.imgix\.net/movie/[^?]+\.jpg(?:\?[^?]+)?',
+                'description': 'md5:b5a8aaeb6993d8ccb18bde8abb8aa8d2',
+                'upload_date': '20231228',
+            },
+        },
+    ]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        # Sets csrftoken cookie
         self._download_webpage(url, video_id)
+        cookies = self._get_cookies(url)
+
+        if 'edu' in url:
+            request_url = f'https://api.medici.tv/edu-satie/edito/movie-file/{video_id}/'
+            source_url = 'https://edu.medici.tv'
+        else:
+            request_url = f'https://api.medici.tv/satie/edito/movie-file/{video_id}/'
+            source_url = 'https://www.medici.tv'
 
-        MEDICI_URL = 'http://www.medici.tv/'
+        mAuth_cookie = cookies.get('auth._token.mAuth')
+        if not mAuth_cookie:
+            token = ''
+        else:
+            token = unquote(str_or_none(mAuth_cookie.value, ''))
```

Comment:
import `try_call` from utils
```suggestion
        token = try_call(lambda: urllib.parse.unquote(self._get_cookies(url)['auth._token.mAuth'].value))
```

Reply:
This one fails to get token